packages/connect-common/src/messageChannel/abstract.ts

```diff
@@ -164,7 +168,7 @@ export abstract class AbstractMessageChannel<
         }
         const messagePromisesLength = Object.keys(this.messagePromises).length;
         if (messagePromisesLength > 5) {
-            this.logger.warn(
+            this.logger?.warn(
```

Comment:
This was not cached by typescript but when reached in cases that it was not defined it was throwing error.